providers/openshift/provider.go

```diff
@@ -29,6 +30,12 @@ import (
 	authorizationv1beta1 "k8s.io/client-go/pkg/apis/authorization/v1beta1"
 )
 
+// httpClientCache stores httpClient objects so that new client does not have to
+// be created on each request just to prevent CAs content changed
+// key is bytes of the hashed metadata of the files for CAs the client was
+// created with
+var httpClientCache sync.Map
```

Comment:
Point out that old entries from this map are never cleaned up.  Up to you if you want to try and fix that.

Reply:
We'd have to be extra careful not to remove a client that some other thread is still using. While not all that clean, I think I'll rather add a note about this.